bucket4j-spring-boot-starter-context/src/main/java/com/giffing/bucket4j/spring/boot/starter/context/constraintvalidations/Bucket4JConfigurationPredicateNameValidator.java

```diff
@@ -0,0 +1,71 @@
+package com.giffing.bucket4j.spring.boot.starter.context.constraintvalidations;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import org.springframework.http.server.reactive.ServerHttpRequest;
+
+import com.giffing.bucket4j.spring.boot.starter.context.ExecutePredicate;
+import com.giffing.bucket4j.spring.boot.starter.context.ExecutePredicateDefinition;
+import com.giffing.bucket4j.spring.boot.starter.context.FilterMethod;
+import com.giffing.bucket4j.spring.boot.starter.context.properties.Bucket4JConfiguration;
+
+public class Bucket4JConfigurationPredicateNameValidator implements ConstraintValidator<ValidPredicateNames, Bucket4JConfiguration> {
```

Comment:
The problem here is, that the user is forced to add the the Servlet and Reactive dependencies to performa the validation. Or am I wrong? The jarkarta servlet dependency is also not scoped as provided so it is everytime added as a dependency. And thats only for the HttpServletRequest. The starter should work only with webflux or with the servlet api.

Reply:
It does require the Jakarta Servlet dependency yes, the Reactive was already included in the library through Spring itself. 
I did try to add the Jakarta dependency as scope provided, but that caused all kinds of exceptions saying the HttpServletRequest class could not be found. Maybe I did something wrong there, I could try it again tomorrow. Or do you perhaps know a fix for this?

Reply:
Yes, the reactive ServerHttpRequest is in the spring-web dependency which will be used for Servlet and Reactive. But HttpServletRequest ist jakarta specific. I don't know how to fix it. With the Validator which uses both Request classes you're forced to add the dependency. I'll think about it again...

Reply:
It took me quite some time to find a solution for this problem.. A simple @ConditionalOnClass would not work since it's not an @Configuration or @Service annotated class, causing the annotations to just be ingore. 

In the end I finally found a solution that does work. Can you let me know what you think of this solution before i merge it and mark the comment as resolved? 
https://github.com/Edwin9292/bucket4j-spring-boot-starter/pull/4

Reply:
👍